<h1>Class PrivilegedActionException</h1>   <ul class="inheritance"> <li><a href="../lang/object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="../lang/throwable">java.lang.Throwable</a></li> <li> <ul class="inheritance"> <li><a href="../lang/exception">java.lang.Exception</a></li> <li> <ul class="inheritance"> <li>java.security.PrivilegedActionException</li> </ul> </li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><a href="../io/serializable">Serializable</a></dd> </dl>   <pre data-language="java">public class PrivilegedActionException
extends Exception</pre> <div>
<p>This exception is thrown by <code>doPrivileged(PrivilegedExceptionAction)</code> and <code>doPrivileged(PrivilegedExceptionAction,
 AccessControlContext context)</code> to indicate that the action being performed threw a checked exception. The exception thrown by the action can be obtained by calling the <code>getException</code> method. In effect, an <code>PrivilegedActionException</code> is a "wrapper" for an exception thrown by a privileged action. </p>
<p>As of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The "exception thrown by the privileged computation" that is provided at construction time and accessed via the <a href="privilegedactionexception#getException--"><code>getException()</code></a> method is now known as the <i>cause</i>, and may be accessed via the <a href="../lang/throwable#getCause--"><code>Throwable.getCause()</code></a> method, as well as the aforementioned "legacy method."</p>
</div> <dl> <dt>See Also:</dt> <dd>
<a href="privilegedexceptionaction"><code>PrivilegedExceptionAction</code></a>, <a href="accesscontroller#doPrivileged-java.security.PrivilegedExceptionAction-"><code>AccessController.doPrivileged(PrivilegedExceptionAction)</code></a>, <a href="accesscontroller#doPrivileged-java.security.PrivilegedExceptionAction-java.security.AccessControlContext-"><code>AccessController.doPrivileged(PrivilegedExceptionAction,AccessControlContext)</code></a>, Serialized Form</dd> </dl>       <h2 id="constructor.detail">Constructors</h2>   <h3 id="PrivilegedActionException-java.lang.Exception-">PrivilegedActionException</h3> <pre data-language="java">public PrivilegedActionException(Exception exception)</pre> <p>Constructs a new PrivilegedActionException "wrapping" the specific Exception.</p> <dl> <dt>Parameters:</dt> <dd>
<code>exception</code> - The exception thrown</dd> </dl>     <h2 id="method.detail">Methods</h2>   <h3 id="getException--">getException</h3> <pre data-language="java">public Exception getException()</pre> <div>
<p>Returns the exception thrown by the privileged computation that resulted in this <code>PrivilegedActionException</code>. </p>
<p>This method predates the general-purpose exception chaining facility. The <a href="../lang/throwable#getCause--"><code>Throwable.getCause()</code></a> method is now the preferred means of obtaining this information.</p>
</div> <dl> <dt>Returns:</dt> <dd>the exception thrown by the privileged computation that resulted in this <code>PrivilegedActionException</code>.</dd> <dt>See Also:</dt> <dd>
<a href="privilegedexceptionaction"><code>PrivilegedExceptionAction</code></a>, <a href="accesscontroller#doPrivileged-java.security.PrivilegedExceptionAction-"><code>AccessController.doPrivileged(PrivilegedExceptionAction)</code></a>, <a href="accesscontroller#doPrivileged-java.security.PrivilegedExceptionAction-java.security.AccessControlContext-"><code>AccessController.doPrivileged(PrivilegedExceptionAction,
                                            AccessControlContext)</code></a>
</dd> </dl>    <h3 id="getCause--">getCause</h3> <pre data-language="java">public Throwable getCause()</pre> <p>Returns the cause of this exception (the exception thrown by the privileged computation that resulted in this <code>PrivilegedActionException</code>).</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="../lang/throwable#getCause--">getCause</a></code> in class <code><a href="../lang/throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>the cause of this exception.</dd> <dt>Since:</dt> <dd>1.4</dd> </dl>    <h3 id="toString--">toString</h3> <pre data-language="java">public String toString()</pre> <p>Description copied from class: <code><a href="../lang/throwable#toString--">Throwable</a></code></p> <p>Returns a short description of this throwable. The result is the concatenation of: <ul> <li> the <a href="../lang/class#getName--">name</a> of the class of this object </li>
<li> ": " (a colon and a space) </li>
<li> the result of invoking this object's <a href="../lang/throwable#getLocalizedMessage--"><code>Throwable.getLocalizedMessage()</code></a> method </li>
</ul> If <code>getLocalizedMessage</code> returns <code>null</code>, then just the class name is returned.</p> <dl> <dt>Overrides:</dt> <dd>
<code><a href="../lang/throwable#toString--">toString</a></code> in class <code><a href="../lang/throwable">Throwable</a></code>
</dd> <dt>Returns:</dt> <dd>a string representation of this throwable.</dd> </dl>     <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993&ndash;2017, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    
  </p>
</div>
